dash/include/dash/algorithm/Accumulate.h

```diff
@@ -41,9 +41,9 @@ ValueType accumulate(
   auto l_first     = index_range.begin;
   auto l_last      = index_range.end;
   auto l_result    = std::accumulate(l_first, l_last, init);
-  auto result      = 0;
+  ValueType result = 0;
```

Comment:
That should be `ValueType result = result` to support non-basic types.

Reply:
With @devreals changes included, we may revert to `auto result = init`? Just a guess.